test/unit/test.database.js

```diff
@@ -1,13 +1,252 @@
+      return Object.assign(buildLog({
+        eventName: "FeeWindowCreated",
+        universe,
+      }, args));
+    },
+    TokensTransferred: args => {
+      required(args, "token", "from", "to", "value", "market");
+      return buildLog({
+        eventName: "TokensTransferred",
+        universe,
+      }, args);
+    },
+    MarketCreated: args => {
+      required(args, "market", "marketCreator");
+      return buildLog({
+        eventName: "MarketCreated",
+        universe,
+        topic: uuid.v4(),
+        description: uuid.v4(),
+        extraInfo: {
+          resolutionSource: "",
+          tags: [],
+        },
+        marketCreationFee: "0.01",  // how much is extracted from trades as market creator
+        minPrice: "0",
+        maxPrice: "1",
+        // See constants.ts/MarketType enum for details.
+        // 0 -> yesNo aka binary
+        // 1 -> categorical aka 3-8 outcomes. Log must include "outcomes".
+        marketType: "0",
+        outcomes: [],  // if marketType=1 then this must be specified
+      }, args);
+    },
+    OrderCreated: args => {
+      required(
+        args,
+        "shareToken",
+        "amount",
+        "price",
+        "creator",
+        "moneyEscrowed",
+        "sharesEscrowed",
+        "tradeGroupId",
+      );
+      return buildLog({
+        eventName: "OrderCreated",
+        universe,
+        orderType: "0",
+        orderId: uuid.v4(),
+      }, args);
+    },
+  });
+}
+
+function makeMockAugur(additional) {
+  const augur = new Augur();
+  augur.api = {
+    Universe: {
+      getReputationToken: () => "REP_TOKEN",
+      getOrCacheReportingFeeDivisor: () => 1,
+    },
+    Market: {
+      getFeeWindow: () => "FEE_WINDOW",
+      getEndTime: () => 592903560,
+      getDesignatedReporter: () => "DESIGNATED_REPORTER",
+      getMarketCreatorMailbox: () => "MARKET_CREATOR_MAILBOX",
+      getNumTicks: () => 42,
+      getMarketCreatorSettlementFeeDivisor: () => 0.01,
+      getValidityBondAttoeth: () => "47",  // must be a number
+      getShareToken: args => Promise.resolve(`SHARE_TOKEN_${args._outcome}`),
+    },
+  };
+  augur.rpc = {
+    getNetworkID: () => "9090",
+  };
+  augur.contracts = {
+    addresses: {
+      9090: {
+      },
+    },
+  };
+
+  return deepmerge(augur, additional || {});
+}
+
+function sortIntegers(ints) {
+  const BASE = 10;
+  // assumes that the ints are strings but also works if they're numbers
+  ints = _.map(ints, n => BigNumber(n, BASE));
+  ints.sort((left, right) => left.comparedTo(right));
+  ints = _.map(ints, n => n.toString(BASE));
+  return ints;
+}
+
+function setupTestDb(augur, logs, blockDetails) {
+  augur = augur || new Augur();
+  logs = logs || [];
+  blockDetails = blockDetails || {};
+
+  const env = getEnv();
   const db = Knex(env);
   return db.migrate.latest(env.migrations)
-    .then(() => db.seed.run(env.seeds))
+    .then(() => {
+      const blockNumbers = sortIntegers(Object.keys(blockDetails));
```

Comment:
`_(blockDetails).keys().map(Number).sortBy().value()`

Not sure if it's better than the function you've written though.

Reply:
Yours is more concise so I'm going with it. Also TIL about `_(someNativeCollection)`!